core/dbt/cli/params.py

```diff
@@ -41,17 +41,10 @@
     default=True,
 )
 
-compile_parse = click.option(
-    "--compile/--no-compile",
-    envvar=None,
-    help="TODO: No help text currently available",
-    default=True,
-)
```

Comment:
It looks like removing this param is what's causing the failure in unit/integration tests, because it's still decorating the `parse` command here:
https://github.com/dbt-labs/dbt-core/blob/137dd9aa1b19222b2f9e8fbae719bca9496f0897/core/dbt/cli/main.py#L393

Given that it no longer _does_ anything, we should just remove it in both places!